src/transformers/models/mobilenet_v1/modeling_mobilenet_v1.py

```diff
@@ -0,0 +1,494 @@
+# coding=utf-8
+# Copyright 2022 Apple Inc. and The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" PyTorch MobileNetV1 model."""
+
+
+from typing import Optional, Union
+
+import torch
+from torch import nn
+from torch.nn import BCEWithLogitsLoss, CrossEntropyLoss, MSELoss
+
+from ...activations import ACT2FN
+from ...modeling_outputs import BaseModelOutputWithPoolingAndNoAttention, ImageClassifierOutputWithNoAttention
+from ...modeling_utils import PreTrainedModel
+from ...utils import add_code_sample_docstrings, add_start_docstrings, add_start_docstrings_to_model_forward, logging
+from .configuration_mobilenet_v1 import MobileNetV1Config
+
+
+logger = logging.get_logger(__name__)
+
+
+# General docstring
+_CONFIG_FOR_DOC = "MobileNetV1Config"
+_FEAT_EXTRACTOR_FOR_DOC = "MobileNetV1FeatureExtractor"
+
+# Base docstring
+_CHECKPOINT_FOR_DOC = "Matthijs/mobilenet_v1_1.0_224"
+_EXPECTED_OUTPUT_SHAPE = [1, 1024, 7, 7]
+
+# Image classification docstring
+_IMAGE_CLASS_CHECKPOINT = "Matthijs/mobilenet_v1_1.0_224"
+_IMAGE_CLASS_EXPECTED_OUTPUT = "tabby, tabby cat"
+
+
+MOBILENET_V1_PRETRAINED_MODEL_ARCHIVE_LIST = [
+    "Matthijs/mobilenet_v1_1.0_224",
+    "Matthijs/mobilenet_v1_0.75_192",
+    # See all MobileNetV1 models at https://huggingface.co/models?filter=mobilenet_v1
+]
+
+
+def _build_tf_to_pytorch_map(model, config, tf_weights=None):
```

Comment:
Nit: this function and the second one would go better in the conversion module IMO.

Reply:
Everyone always copies it from BERT, but it's to load a TensorFlow 1 checkpoint inside a PyTorch model only. So I'm not sure it applies here. The checkpoints you have are TF2 checkpoints, no?

Reply:
They're TF1 or at least the compat.v1 stuff. These models were originally implemented with the old "slim" wrapper that Google used for a while.

Reply:
In this case, the function will need to stay here indeed!